cogito/check.go

```diff
@@ -19,16 +19,35 @@ import (
 // the requested version if it is still valid.
 //
 func Check(log hclog.Logger, in io.Reader, out io.Writer, args []string) error {
-	ci, err := NewCheckInput(in)
-	if err != nil {
-		return fmt.Errorf("check: %s", err)
+	var ci CheckInput
+	dec := json.NewDecoder(in)
+	dec.DisallowUnknownFields()
+	if err := dec.Decode(&ci); err != nil {
+		return fmt.Errorf("check: parsing JSON from stdin: %s", err)
 	}
+	ci.Env.Fill()
 
+	if err := ci.Source.ValidateLog(); err != nil {
```

Comment:
The complication: we need to validate the source configuration twice: now (line 30), so that we can print debug info, and then at line 42 below.